internal/core/application/wallet_service.go

```diff
@@ -151,6 +152,9 @@ func (w *walletService) InitWallet(
 				return nil, err
 			}
 
+			// We unlock the vault setting the mnemonic in memory needed for restore
+			config.Set(config.MnemonicKey, strings.Join(mnemonic, " "))
```

Comment:
We could use vault method to unlock

```
if err = v.Unlock(passphrase); err != nil {
      return nil, err
}
```

Reply:
+1

Reply:
Ok, I think I understood how we introudeced this bug. In this [commit](https://github.com/TDex-network/tdex-daemon/commit/200e6de2a169a5cfe303e0f5b386c76dfdcffc00#diff-872797e6e5246bb52e67bacdbd84b1ce2601e1e20a9e761454ceb4b9c59bf32f) we removed the Mnemonic filed from the Vault struct, and this is ok, but in the NewVault factory function we forgot to call config.Set, like done in Unlock, to store the plain text mnemonic into viper.
Shall we add this into NewVault instead of manually setting it as you've done?

Reply:
Yes please do

Reply:
Could you do in the fix with #185 maybe @altafan